yt_dlp/extractor/startrek.py

```diff
@@ -0,0 +1,59 @@
+from yt_dlp.utils import int_or_none
+from .common import InfoExtractor
+
+
+class StarTrekIE(InfoExtractor):
+    _VALID_URL = r'(?P<base>https://(?:intl|www)\.startrek\.com)/videos/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://intl.startrek.com/videos/watch-welcoming-jess-bush-to-the-ready-room',
+        'md5': '491df5035c9d4dc7f63c79caaf9c839e',
+        'info_dict': {
+            'id': 'watch-welcoming-jess-bush-to-the-ready-room',
+            'ext': 'mp4',
+            'title': 'WATCH: Welcoming Jess Bush to The Ready Room',
+            'duration': 1888,
+            'timestamp': 1655388000,
+            'upload_date': '20220616',
+            'description': 'md5:1ffee884e3920afbdd6dd04e926a1221',
+            # Thumbnail seems to have some tracking parameters added, ignore those.
+            'thumbnail': r're:https://intl\.startrek\.com/sites/default/files/styles/video_1920x1080/public/images/2022-06/pp_14794_rr_thumb_107_yt_16x9.jpg(?:\?.+)?',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        urlbase = self._match_valid_url(url).group('base')
+
+        description = self._html_search_regex(
+            r'(?s)<div class="header-body">(.+?)</div>',
+            webpage, 'description', fatal=False)
+        ld = self._search_json_ld(webpage, video_id, fatal=False)
+        self.write_debug(ld)
+
+        # Extract the player element so that the rest of the matching will be done only there.
+        player = self._search_regex(r'(<div id="cvp-player-.+?></div>)', webpage, 'player')
+
+        hls = self._html_search_regex(r' data-hls="(.+?)" ', player, 'HLS URL')
+        title = self._html_search_regex(r' data-title="(.+?)" ', player, 'title', ld.get('title'))
+        duration = int_or_none(
+            self._html_search_regex(r' data-duration="([0-9]+?)" ', player, 'duration', None))
+        poster = self._html_search_regex(r' data-poster-url="(.+?)" ', player, 'thumbnail', None)
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(hls, video_id)
+        self._sort_formats(formats)
+
+        # Usually, these are just an absolute path. Add the host.
+        if poster is not None and poster.startswith("/"):
+            poster = urlbase + poster
+
+        return {
+            'id': video_id,
+            'title': title,
+            'description': description,
+            'duration': duration,
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnail': poster,
+            'timestamp': ld.get('timestamp'),
```

Comment:
You are throwing away all other metadata from json/ld?

Reply:
There’s not much in there:

```python
{
  'title': 'WATCH: Welcoming Jess Bush to The Ready Room',
  'description': 'A closer look at the latest episode of Strange New Worlds',
  'thumbnails': [{'url': 'https://intl.startrek.com/sites/default/files/styles/video_1280x720/public/images/2022-06/pp_14794_rr_thumb_107_yt_16x9.jpg?h=d1cb525d&amp;itok=pk5UgKFK'}],
  'timestamp': 1655388000,
  'uploader': 'StarTrek.com Staff'
}
```

* `title` is already extracted from the player’s `data-title` (but will be used as a fallback).
* `description` is different from the HTML description below the player and less detailed.
* `thumbnails` are extracted from `data-poster-url`. For the sake of simplicity I was leaning to not using the JSON-LD ones as a fallback either (because there can be multiple, while `data-poster-url` only has one). Also, I expect `data-poster-url` to be of the same or higher quality than the JSON-LD ones, because `data-poster-url` is what’s being used in the (large) player, while the JSON-LD one(s) are more likely to be smaller ones just for SEO & social media. If you disagree, let me know.
* `timestamp` is used.
* `uploader` does not look meaningful and is not shown to the user. Let me know if I should add it nevertheless.

Reply:
Also, now that I look at it more closely: Is `&amp;` appearing in the thumbnail URL value a bug in `_search_json_ld()`? I’ve checked whether it’s accidentally double-encoded on the page; it isn’t.

Reply:
> Also, now that I look at it more closely: Is &amp; appearing in the thumbnail URL value a bug in _search_json_ld()? I’ve checked whether it’s accidentally double-encoded on the page; it isn’t.

Possibly. I'll check